packages/devtools_app/test/code_size_screen_test.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@TestOn('vm')
+import 'package:devtools_app/src/code_size/code_size_screen.dart';
+import 'package:devtools_app/src/code_size/code_size_controller.dart';
+import 'package:devtools_app/src/split.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'support/wrappers.dart';
+
+void main() {
+  CodeSizeScreen screen;
+  CodeSizeController snapshotTabController;
+
+  Future<void> pumpCodeSizeScreen(
+    WidgetTester tester, {
+    CodeSizeController codeSizeController,
+  }) async {
+    await tester.pumpWidget(wrapWithControllers(
+      const CodeSizeBody(),
+      codeSize: codeSizeController,
+    ));
+    await tester.pumpAndSettle(const Duration(seconds: 1));
+    expect(find.byType(CodeSizeBody), findsOneWidget);
+  }
+
+  const windowSize = Size(2050.0, 1000.0);
+
+  group('CodeSizeScreen', () {
+    setUp(() async {
+      screen = const CodeSizeScreen();
+      snapshotTabController = CodeSizeController();
+      await snapshotTabController.loadTree(
+        '../devtools_testing/lib/support/treemap_test_data_v8_new.json',
+      );
+    });
+
+    testWidgets('builds its tab', (WidgetTester tester) async {
+      await tester.pumpWidget(wrapWithControllers(
+        Builder(builder: screen.buildTab),
+        codeSize: snapshotTabController,
+      ));
+      expect(find.text('Code Size'), findsOneWidget);
+    });
+
+    testWidgetsWithWindowSize('builds tabs', windowSize,
+        (WidgetTester tester) async {
+      await pumpCodeSizeScreen(
+        tester,
+        codeSizeController: snapshotTabController,
+      );
+
+      expect(snapshotTabController.currentRoot.value, isNotNull);
+      expect(find.byType(CodeSizeBody), findsOneWidget);
+      expect(find.byType(TabBar), findsOneWidget);
+
+      const snapshotTabKey = Key('Code Size Snapshot Tab');
```

Comment:
instead of creating the keys here, grab them directly from the widget where they are declared
`CodeSizeBodyState.snapshotTabKey`